UnityProject/Assets/Scripts/Player/PlayerSync.Server.cs

```diff
@@ -1,982 +1,1038 @@
+        {
+            if (crossedItem.HasTrait(CommonTraits.Instance.Slippery))
+            {
+                registerPlayer.ServerSlip(slipWhileWalking: true);
+            }
+        }
+    }
+
+    public void CheckTileSlip()
+    {
+
+        var matrix = MatrixManager.Get(serverState.MatrixId);
+
+        var shoeSlot = playerScript.ItemStorage.GetNamedItemSlot(NamedSlot.feet);
+
+        bool slipProtection = !shoeSlot.IsEmpty && shoeSlot.ItemAttributes.HasTrait(CommonTraits.Instance.NoSlip);
+
+        if (matrix.MetaDataLayer.IsSlipperyAt(ServerLocalPosition) && !slipProtection)
+        {
+            registerPlayer.ServerSlip();
+        }
+    }
+
+    public void CheckTileContagion()
+    {
+    }
+
+    /// <summary>
+    /// Play footsteps at given position. It will handle all the logic to determine
+    /// the proper sound to use.
+    /// </summary>
+    /// <param name="worldPos">Where in the world is this sound coming from. Also used to get the type of tile</param>
+    /// <param name="stepType">What kind of step does the creature walking have</param>
+    /// <param name="performer">The creature making the sound</param>
+    public async void FootstepAtPosition(Vector3 worldPos, StepType stepType, GameObject performer)
+    {
+        MatrixInfo matrix = MatrixManager.AtPoint(worldPos.NormalizeToInt(), false);
+        var locPos = matrix.ObjectParent.transform.InverseTransformPoint(worldPos).RoundToInt();
+        var tile = matrix.MetaTileMap.GetTile(locPos) as BasicTile;
+
+        if (tile != null)
+        {
+            if (step)
+            {
+                AudioSourceParameters audioSourceParameters = new AudioSourceParameters
+                {
+                    Pitch = UnityEngine.Random.Range(0.7f, 1.2f)
+                };
+
+                StepTypes stepTypes = playerMove.StepSounds.StepTypes.FirstOrDefault(p => p.StepType == stepType);
+
+                if (stepTypes == null || stepTypes.FloorTileTypes.Count == 0)
+                {
+                    Logger.LogWarning($"No step sound for step type {stepType}");
+                    return;
+                }
+
+                FloorTileTypes floorTileTypes = stepTypes.FloorTileTypes.FirstOrDefault(p => p.FloorTileType == tile.floorTileType);
+
+                if (floorTileTypes == null || floorTileTypes.AddressableAudioSources == null || floorTileTypes.AddressableAudioSources.Count == 0)
+                {
+                    Logger.LogWarning($"No step sound for floor tile type {tile.floorTileType}");
+                    return;
+                }
+
+                await SoundManager.PlayNetworkedAtPos(
+                    floorTileTypes.AddressableAudioSources,
+                    worldPos,
+                    audioSourceParameters,
+                    polyphonic: true,
+                    Global: false,
+                    sourceObj: performer
+                );
+            }
+
+            step = !step;
```

Comment:
Codacy found an issue: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)